internal/handlers/common/update.go

```diff
@@ -18,18 +18,175 @@ import (
 	"fmt"
 	"sort"
 	"strings"
+	"time"
 
 	"github.com/FerretDB/FerretDB/internal/types"
 	"github.com/FerretDB/FerretDB/internal/util/must"
 )
 
+// CheckCurrentdateFieldExpression checks $currentDate input on correctness.
+func CheckCurrentdateFieldExpression(update *types.Document) error {
+	for _, updateOp := range update.Keys() {
+		if updateOp != "$currentDate" {
```

Comment:
I need my PR for `$setOnInsert` here so when it will be merged, this code will change also.